src/main/java/org/zalando/nakadi/service/TopicsService.java

```diff
@@ -0,0 +1,61 @@
+package org.zalando.nakadi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.config.SecuritySettings;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.Timeline;
+import org.zalando.nakadi.exceptions.ForbiddenAccessException;
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.exceptions.NakadiRuntimeException;
+import org.zalando.nakadi.exceptions.runtime.NoTopicFoundException;
+import org.zalando.nakadi.repository.EventTypeRepository;
+import org.zalando.nakadi.repository.db.TimelineDbRepository;
+import org.zalando.nakadi.security.Client;
+
+import java.util.Optional;
+
+@Component
+public class TopicsService {
+
+    private final SecuritySettings securitySettings;
+    private final TimelineDbRepository timelineDbRepository;
+    private final EventTypeRepository eventTypeRepository;
+
+    @Autowired
+    public TopicsService(final SecuritySettings securitySettings,
+                         final TimelineDbRepository timelineDbRepository,
+                         final EventTypeRepository eventTypeRepository) {
+        this.securitySettings = securitySettings;
+        this.timelineDbRepository = timelineDbRepository;
+        this.eventTypeRepository = eventTypeRepository;
+    }
+
+    public EventType getEventTypeByTopic(final String topic, final Client client) {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        final Optional<String> etName = timelineDbRepository.listTimelinesOrdered().stream()
```

Comment:
I understand that this operation is really rare, but you will fetch all Timelines in db. At some point nakadi will just die.